projects/core/src/occ/adapters/site-context/converters/base-site-normalizer.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+import { BaseSite } from '../../../../model/misc.model';
+import { Converter } from '../../../../util/converter.service';
+import { Occ } from '../../../occ-models/occ.models';
+import { JavaRegExpConverter } from './java-reg-exp-converter';
+
+@Injectable({ providedIn: 'root' })
+export class BaseSiteNormalizer implements Converter<Occ.BaseSite, BaseSite> {
+  constructor(protected javaRegExpConverter: JavaRegExpConverter) {}
```

Comment:
Only now I tested and learned that `TransferState` cannot transmit `RegExp` objects from SSR to CSR via JSON (`JSON.stringify()` on `RegExp` returns `{}`).

I'd like to avoid adding custom mechanism for converting RegExp to string in SSR and back from string to RegExp in CSR. It's tricky.

So now I'd like to have in ngrx store the original raw Java regexes, but convert them to JS RegExp objects later.

Reply:
Done